### PR TITLE
Maintenance

### PR DESCRIPTION
**Pull request by mikesaelim**
- Bump the ruby version used in development to 3.0.0
- Add ruby 2.7.2 and 3.0.0 to the Travis build matrix, and remove ruby 2.3 and 2.4.
- Fix deprecation warnings around [positional and keyword arguments in ruby 2.7.0](https://www.ruby-lang.org/en/news/2019/12/25/ruby-2-7-0-released/)
- Upgrade rubocop to the latest version, v1.9.1, and fix rubocop violations
- Configure rspec to emit warnings, and fix them
- Re-record the spec cassettes, which are almost 4 years old 😅 
- Fix broken links in the README

There are unfixed warnings in the ruby 2.6+ builds caused by our use of positional arguments when calling `YAML.safe_load`.  But we cannot change them to keyword arguments until we drop support for ruby 2.5, because those keyword arguments aren't available until ruby 2.6.  I've filed an issue for later: #13.

### CODE REVIEW

**Review by mattyb — DISMISSED** [review]
LGTM

**Comment by mikesaelim** [comment]
@mattyb I still have yet to update the version and changelog files, but first I wanted your opinion on these changes, and on whether this should be a minor or major version bump.  Does dropping support for ruby 2.3 and 2.4 count as a backwards incompatibility, even though both are quite past end-of-life?

Also, should I move the re-recording of the spec cassettes into another PR to keep this one cleaner?

File: .rubocop.yml [hunk 1]
```diff
@@ -1,5 +1,7 @@
 AllCops:
-  TargetRubyVersion: 2.3
+  TargetRubyVersion: 2.5
+  NewCops: disable
```
Comment (mikesaelim) [inline .rubocop.yml:3]: https://docs.rubocop.org/rubocop/1.9/versioning.html#pending-cops

Rubocop currently passes all the new cops too, but I didn't want to open us up to being annoyed by the new cops every time we upgrade rubocop, and I didn't feel like enabling all the new cops individually in this file.  What do you think the best option is?

File: mistral_client.gemspec [hunk 2]
```diff
@@ -14,7 +14,7 @@ Gem::Specification.new do |spec|
   spec.homepage      = 'https://github.com/civisanalytics/mistral_client'
   spec.license       = 'BSD 3-Clause'
 
-  spec.required_ruby_version = '~> 2.3'
+  spec.required_ruby_version = '>= 2.5'
```
Comment (mikesaelim) [inline mistral_client.gemspec]: Or should I do something like `['>= 2.5', '< 3.1']`?

Comment (mattyb) [reply mistral_client.gemspec]: Yeah, minor versions can have incompatible changes, so it probably does make sense to specify what we've tested on

**Comment by mattyb** [comment]
> Does dropping support for ruby 2.3 and 2.4 count as a backwards incompatibility, even though both are quite past end-of-life?
> 
> Also, should I move the re-recording of the spec cassettes into another PR to keep this one cleaner?

this all looks good! I do think we should bump the major version if we're dropping support for ruby versions. I think it's fine to keep the cassettes in this PR, unless there are related code fixes (besides that URL change) that start to complicate things.

**Review by mattyb — APPROVED** [review]
LGTM!

**Pull request merged**